aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -50,15 +50,18 @@ Tensor sparse_csr_tensor(
       options.layout() == kSparseCsr,
       "expected sparse CSR layout, but got layout ",
       options.layout());
-
-  AT_DISPATCH_INDEX_TYPES(crow_indices.scalar_type(), "csr_construct_check", [&] {
-    auto crow_indices_accessor = crow_indices.accessor<index_t, 1>();
-    TORCH_CHECK(
-        crow_indices_accessor[crow_indices.numel() - 1] <= col_indices.numel(),
-        "last value of crow_indices should be less than length of col_indices.");
-    TORCH_CHECK(
-        crow_indices_accessor[0] == 0, "0th value of crow_indices must be 0.");
-  });
+  if (crow_indices.is_cpu()) {
+    AT_DISPATCH_INDEX_TYPES(crow_indices.scalar_type(), "csr_construct_check", [&] {
+      auto crow_indices_accessor = crow_indices.accessor<index_t, 1>();
+      TORCH_CHECK(
+          crow_indices_accessor[crow_indices.numel() - 1] <= col_indices.numel(),
+          "last value of crow_indices should be less than length of col_indices.");
+      TORCH_CHECK(
+          crow_indices_accessor[0] == 0, "0th value of crow_indices must be 0.");
+    });
+  } else {
+    //@aocsa, TODO: cuda case
```

Comment:
TODO

Reply:
It seems like you're going to need a sync here. Is it possible some of this information should be stored on CPU, or perhaps maybe you should just do an async cuda assert here.